var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -72,16 +75,20 @@ def configure_args(self):
 
         args = [
             '--with-zlib=' + spec['zlib'].prefix,
-            '--with-libidn2=' + spec['libidn2'].prefix,
             # Prevent unintentional linking against system libraries: we could
             # add variants for these in the future
-            '--without-libbrotli',
+            '--without-brotli',
             '--without-libgsasl',
             '--without-libmetalink',
             '--without-libpsl',
             '--without-zstd',
         ]
 
+        if spec.satisfies('+libidn2'):
+            args.append('--with-libidn2=' + spec['libidn2'].prefix)
+        else:
+            args.append('--without-libidn2')
```

Comment:
can you replace this with:

```python
args += self.with_or_without(libidn2, 'prefix')
```

See https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#helper-functions

Reply:
done